product_pricelist_fixed_price/model/product_pricelist_item.py

```diff
@@ -1,50 +1,124 @@
+        """Make sure that after updating database tables for this module,
+        existing values in pricelist item are set correctly."""
+        cr.execute(
+            'update product_pricelist_item'
+            ' set base_ext = base'
+            ' where base_ext != -3 and base != base_ext'
+        )
+        return super(ProductPricelistItem, self)._auto_end(cr, context=context)
+
+    @api.model
+    def _get_fixed_price_base(self):
+        """Return value to set on base when using fixed price item."""
+        base = self.env['product.price.type'].search([], limit=1, order='id')
+        assert base, _("No record found in product.price.type")
+        return base[0].id
+
+    @api.model
+    def _modify_vals(self, vals):
+        """Ensure consistent values for fixed pricelist items.
+
+        The passed vals parameter is used for both input and output.
+        base should be default if base-ext = -1, in all other cases base and
+        base_ext should be the same.
+        """
+        # Check wether any action is needed
+        if not ('base_ext' in vals or 'base' in vals):
+            return
+        # Get base and base_ext values
+        if 'base_ext' in vals:
+            base_ext = vals['base_ext']
+            if base_ext != FIXED_PRICE_TYPE:
+                base = base_ext
+            else:
+                base = self._get_fixed_price_base()
+        else:
+            # getting here we are sure base is in vals
+            base = vals['base']
+            base_ext = base
+        # Synchronize base and base_ext values
+        vals.update({
+            'base_ext': base_ext,
+            'base': base,
+        })
+        # Make sure other values valid for fixed price
+        if base_ext == FIXED_PRICE_TYPE:
+            vals.update({
+                'price_discount': -1.0,
+                'price_round': 0.0,
+                'price_min_margin': 0.0,
+                'price_max_margin': 0.0,
+            })
+
+    @api.model
+    def create(self, vals):
+        """override create to get computed values"""
+        self._modify_vals(vals)
+        return super(ProductPricelistItem, self).create(vals)
+
+    @api.multi
+    def write(self, vals):
+        """override write to get computed values."""
+        self._modify_vals(vals)
+        return super(ProductPricelistItem, self).write(vals)
 
     @api.onchange('base_ext')
     def change_base_ext(self):
-        base = self.base_ext
-        if base == -3:
-            base = self._get_default_base(
-                {'type': self.price_version_id.pricelist_id.type})
+        if self.base_ext == FIXED_PRICE_TYPE:
+            self.base = self._get_fixed_price_base()
             self.price_discount = -1
-        self.base = base
+            self.price_round = 0.0
+            self.price_min_margin = 0.0
+            self.price_max_margin = 0.0
+        else:
+            self.base = self.base_ext
+            if self.price_discount == -1.0:
```

Comment:
Would it not be better to let the user to decide in this case ? Maybe his change does not come from the value `-3` keeping the formula valid without forcing `discount=0`... Maybe his change is coming from the value `-3` making the formula highly suspect by forcing `discount=0` if the user does not touch also initial `surcharge`